pkg/portal/portal.go

```diff
@@ -261,21 +265,41 @@ func (p *portal) unauthenticatedRoutes(r *mux.Router) {
 }
 
 func (p *portal) aadAuthenticatedRoutes(r *mux.Router) {
-	for _, name := range AssetNames() {
-		if name == "v1/build/index.html" {
-			r.NewRoute().Methods(http.MethodGet).Path("/").HandlerFunc(p.index)
-			continue
-		}
+	var names []string
 
-		if name == "v2/build/index.html" {
-			r.NewRoute().Methods(http.MethodGet).Path("/v2").HandlerFunc(p.indexV2)
-			continue
+	err := filepath.Walk(".", func(path string, info fs.FileInfo, err error) error {
+		if err != nil {
```

Comment:
This error check doesn't fully make sense to me.

`Walk` runs this function on ever file it identifies, right? So nothing in here should have errored yet if I understand correctly.

Reply:
@cblecker - Thanks for your review feedback. 

`Walk` just walks the file tree and calling `WalkFunc` for each file or directory in the tree. All errors that arise visiting files and directories are filtered by `WalkFunc`. I just use the standard error handling of this function to handle the possible failures accessing a path. 

This is an example [example of using Walk](https://pkg.go.dev/path/filepath#Walk) I referred. Please let me know if anything I misunderstood from you.

Reply:
the error is given by walk to the function. We have to check it and decide how we handle errors and if we should stop the whole processing or just skip the directory etc. Maybe crashing the portal here isn't optimal though? I'd rather keep what they have in the example, which will stop the walking and we can handle the error in the following block.

Reply:
I would also prefer if we took out the function into a real function instead of using an anonymous function here.

Reply:
It's the standard use case of the `Walk` function to have an anonymous `WalkFunc` function type called by Walk. May I know the reason it's really needed to take out into a real function?